sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -868,8 +868,8 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase with ExecutionE
   def unsupportedPartitionTransformError(
       transform: Transform): SparkUnsupportedOperationException = {
     new SparkUnsupportedOperationException(
-      errorClass = "_LEGACY_ERROR_TEMP_2067",
-      messageParameters = Map("transform" -> transform.toString()))
+      errorClass = "UNSUPPORTED_PARTITION_TRANSFORM",
+      messageParameters = Map("transform" -> toSQLId(transform.toString)))
```

Comment:
Isn't `transform` an expression actually. If so, please, use `toSQLExpr`

Reply:
I tried it but unfortunately it says `Transform` is not an `Expression`:

<img width="655" alt="Screenshot 2024-10-29 at 1 54 45 PM" src="https://github.com/user-attachments/assets/8f58f8a9-9b2a-4321-a5ec-162cef807c35">

Is there happen to any way converting `Transform` to `Expression`?

Reply:
ah, I see. `Transform` extends another `Expression` from `org.apache.spark.sql.connector.expressions`. Could you add one more definition of `toSQLExpr` which accept `connector.expressions.Expression`. It should quote in the same way as the existing `toSQLExpr` (using `quoteByDefault`).

Reply:
Sure will add it. Thanks for the comment!

Reply:
I see. thank you for the investigation.